model/roles.go

```diff
@@ -32,18 +49,100 @@ type Role struct {
 	EnvironScripts    []string       `yaml:"environment_scripts"`
 	Scripts           []string       `yaml:"scripts"`
 	PostConfigScripts []string       `yaml:"post_config_scripts"`
-	Type              string         `yaml:"type,omitempty"`
+	Type              RoleType       `yaml:"type,omitempty"`
 	JobNameList       []*roleJob     `yaml:"jobs"`
-	Configuration     *configuration `yaml:"configuration"`
+	Configuration     *Configuration `yaml:"configuration"`
+	Run               *RoleRun       `yaml:"run"`
+	Tags              []string       `yaml:"tags"`
 
 	rolesManifest *RoleManifest
 }
 
+// RoleRun describes how a role should behave at runtime
+type RoleRun struct {
+	Scaling           *RoleRunScaling       `yaml:"scaling"`
+	Capabilities      []string              `yaml:"capabilities"`
+	PersistentVolumes []*RoleRunVolume      `yaml:"persistent-volumes"`
+	SharedVolumes     []*RoleRunVolume      `yaml:"shared-volumes"`
+	Memory            int                   `yaml:"memory"`
+	VirtualCPUs       int                   `yaml:"virtual-cpus"`
+	ExposedPorts      []*RoleRunExposedPort `yaml:"exposed-ports"`
+	FlightStage       FlightStage           `yaml:"flight-stage"`
+	HealthCheck       *HealthCheck          `yaml:"healthcheck,omitempty"`
+}
+
+// RoleRunScaling describes how a role should scale out at runtime
+type RoleRunScaling struct {
+	Min int32 `yaml:"min"`
+	Max int32 `yaml:"max"`
+}
+
+// RoleRunVolume describes a volume to be attached at runtime
+type RoleRunVolume struct {
+	Path string `yaml:"path"`
+	Tag  string `yaml:"tag"`
+	Size int    `yaml:"size"`
+}
+
+// RoleRunExposedPort describes a port to be available to other roles, or the outside world
+type RoleRunExposedPort struct {
+	Name     string `yaml:"name"`
+	Protocol string `yaml:"protocol"`
+	External string `yaml:"external"`
+	Internal string `yaml:"internal"`
+	Public   bool   `yaml:"public"`
+}
+
+// HealthCheck describes a non-standard health check endpoint
+type HealthCheck struct {
+	URL     string            `yaml:"url"`     // URL for a HTTP GET to return 200~399. Cannot be used with other checks.
+	Headers map[string]string `yaml:"headers"` // Custom headers; only used for URL.
+	Command []string          `yaml:"command"` // Custom command. Cannot be used with other checks.
+	Port    int32             `yaml:"port"`    // Port for a TCP probe. Cannot be used with other checks.
+}
+
 // Roles is an array of Role*
 type Roles []*Role
 
-type configuration struct {
-	Templates map[string]string `yaml:"templates"`
+// Configuration contains information about how to configure the
+// resulting images
+type Configuration struct {
+	Templates map[string]string        `yaml:"templates"`
+	Variables []*ConfigurationVariable `yaml:"variables"`
```

Comment:
Not sure what you mean about pattern (A pattern of how we use Go?). I asked because I saw later (line 123) the definition of `type ConfigurationVariableSlice []*ConfigurationVariable` and wondered if it would make sense to use the type name instead of replicating the definition